src/Livewire/Portal/Shop/ProductDetail.php

```diff
@@ -20,11 +24,26 @@ class ProductDetail extends Component
 
     public array $groups = [];
 
+    #[Url]
+    public ?int $variant = null;
+
     public function mount(Product $product): void
     {
         $product->loadMissing(['children', 'parent']);
 
         $this->fillProductForm($product);
+
+        if ($this->variant) {
+            $variant = app(Product::class)
+                ->query()
+                ->where('parent_id', $product->id)
+                ->whereKey($this->variant)
+                ->with('productOptions:id,product_option_group_id')
+                ->sole();
+            $this->groups = $variant->productOptions->pluck('id', 'product_option_group_id')->toArray();
```

Comment:
this does not work as intended.
The array is set up with ['product_option_group_id' => 'id'], and therefore if multiple product options have the same product_option_group_id only the last one is shown.